plugins/woocommerce/client/admin/client/marketplace/components/header-account/header-account.tsx

```diff
@@ -109,20 +124,28 @@ export default function HeaderAccount(): JSX.Element {
 							<MenuItem
 								className="woocommerce-marketplace__menu-item"
 								href={ accountOrConnect }
+								onClick={ () =>
+									recordEvent(
+										'header_account_connect_click',
+										{ page }
+									)
+								}
 							>
 								{ connectionDetails() }
 							</MenuItem>
-							<MenuItem href={ accountURL }>
-								<Icon
-									icon={ external }
-									size={ 24 }
-									className="woocommerce-marketplace__menu-icon"
-								/>
-								{ __(
-									'WooCommerce.com account',
-									'woocommerce'
-								) }
-							</MenuItem>
+							{ page === 'wc-addons' && (
+								<MenuItem href={ accountURL }>
+									<Icon
+										icon={ external }
+										size={ 24 }
+										className="woocommerce-marketplace__menu-icon"
+									/>
+									{ __(
+										'WooCommerce.com account',
+										'woocommerce'
```

Comment:
I don't think it's necessary to show this once we're already connected, as it duplicates the link attached to the email address.